Fix #416 - crash after adding new account

hiddentao opened the pull request:
Fixes #416. This replaces underscore with lodash (as this provides for improved functionality) and fixes the crash which occurs after adding a new account.

hiddentao commented:
Ah, sorry I'd accidentally merged #393 into my `master` If we're not going to merge that one in I can redo this PR against official `master`.

frozeman commented:
I will take a look, but please always make PRs against develop only

frozeman commented on README.md:
```diff
@@ -37,7 +37,7 @@ Requires electron version 0.37.2
 
     $ curl https://install.meteor.com/ | sh
-    $ npm install -g electron-prebuilt
+    $ npm install -g electron-prebuilt@0.36.9
```
we are already at 0.37.2

frozeman commented on modules/ipcCommunicator.js:
```diff
@@ -4,6 +4,7 @@ Window communication
 @module ipcCommunicator
 */
 
+const _ = require('lodash');
```
There are differences in the API, are you sure it doesn't break anything?
I mostly use `_.each` and `_.isEmpty`, which i saw exist as well.

My personal opinion is, yes it does have performance improvements, but for the use cases were we use it, it makes 0 difference..

frozeman commented on modules/ipcCommunicator.js:
```diff
@@ -34,7 +35,7 @@ ipc.on('backendAction_closeApp', function() {
     var windowId = e.sender.getId();
 
-    if(global.windows[windowId]) {
+    if(_.get(global.windows, windowId)) {
```
I didn't know about `_.get()` seems like a neat function. Though for this use case its probably a bit to much ;)

hiddentao commented:
@frozeman All good points. Initially the code rewrite looked different, which is why I had put lodash in. Then I refactored it and though it doesn't really require lodash like you said I decided to leave it in since it's a better lib. I tried to ensure I replaced all usage instances of underscore with lodash, but without tests it's hard to say.

All in all, I think I'll redo this PR from scratch against `develop`. I should also then redo the other PR I have pending against `develop`.

hiddentao commented:
Replaced by #442
